openwisp_users/api/permissions.py

```diff
@@ -0,0 +1,65 @@
+from django.utils.translation import gettext_lazy as _
+from rest_framework.permissions import BasePermission
+from swapper import load_model
+
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class BaseOrganizationPermission(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        organization = self.get_object_organization(view, obj)
+        return self.validate_membership(request.user, organization)
+
+    def has_permission(self, request, view):
+        return request.user and request.user.is_authenticated
+
+    def get_object_organization(self, view, obj):
+        organization_field = getattr(view, 'organization_field', 'organization')
+        fields = organization_field.split('__')
+        accessed_object = obj
+        for field in fields:
+            accessed_object = getattr(accessed_object, field, None)
+            if not accessed_object:
+                self.message = _(
+                    'Organization not found, `organization_field` '
+                    'not implemented correctly.'
+                )
+                break
+        return accessed_object
+
+    def validate_membership(self, user, org):
+        self.message = _(
```

Comment:
this should raise [NotImplementedError](https://docs.python.org/3/library/exceptions.html#NotImplementedError) with this message instead of trying to display it to the user

Reply:
Ah, that's what I was doing before, but then I thought, letting user know. I've undone that.